samples/error.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2018, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @see [Cloud Datastore Concepts errors] {@link https://cloud.google.com/datastore/docs/concepts/errors}
+ * for more information on errors
+ */
+
+'use strict';
+
+const {Datastore} = require('@google-cloud/datastore');
+
+// [START error]
+function runQuery() {
+  // Creates a client
+  const datastore = new Datastore();
+
+  const query = datastore.createQuery(['Company']).start('badrequest');
+
+  return datastore

```

Comment:
we tend to use an `async` function called `main` these days, here's a sample from the guidelines document:

```js
function main(arg1 = 'default_value_one', arg2 = 1234) {
  // [START your_region_tag]
  /**
   * TODO(developer): Uncomment these variables before running the sample.
   */
  // const arg1 = 'default_value_one';
  // const arg2 = 1234;

  // Imports the Google Cloud Some API library
  const {SomeApiClient} = require('@google-cloud/some-api');

  // Instantiates a client
  const someApiClient = new SomeApiClient();

  async function doSomething() {
    ...
  }

  doSomething();
  // [END your_region_tag]
}

main(...process.argv.slice(2));
```